### PR TITLE
Implement text formatting functions and lists

### PR DESCRIPTION
**Pull request by TerrenceHung**
This PR implements the functions that perform basic text formatting options, as well as unordered/ordered lists.

Each function was also tested by the output into a GitHub issue and previewing the contents to verify they appear correctly.

### CODE REVIEW

File: githubmarkdownui/blocks/lists.py [hunk 1]
```diff
@@ -10,6 +10,18 @@ class HtmlList(ABC):
     """
     items: List[Union[str, HtmlList]]
 
+    def build_list_contents(self) -> str:
+        """Constructs the contents of the list by building the <li> tags or any nested lists."""
+        list_body = ''
+
+        for item in self.items:
+            if isinstance(item, HtmlList):
+                list_body += str(item)
```
Comment (zac-li) [inline githubmarkdownui/blocks/lists.py:21]: so this isn't recursively calling `build_list_contents`?

File: githubmarkdownui/blocks/leaf.py [hunk 2]
```diff
@@ -25,4 +25,7 @@ def heading(text: str, level: int) -> str:
 
     :raises: Exception if the level is not between 1 to 6 inclusive
     """
-    pass
+    if not 1 <= level <= 6:
+        raise Exception('Level must be between 1 to 6 inclusive')
```
Comment (adverma98) [inline githubmarkdownui/blocks/leaf.py]: I believe this is slightly more grammatically sound? I could be wrong though.
```suggestion
        raise Exception('Level must be between 1 and 6 inclusive')
```

File: githubmarkdownui/blocks/lists.py (hunk 1 above)
Comment (TerrenceHung) [reply githubmarkdownui/blocks/lists.py:21]: It's implicitly called through the `str(item)` call. We need the `<li>` or `<ol>` tags of the inner list here to format the list properly in HTML, and when we cast the list the `__str__` function will call `build_list_contents()`

File: githubmarkdownui/blocks/leaf.py (hunk 2 above)
Comment (TerrenceHung) [reply githubmarkdownui/blocks/leaf.py]: I think you're right, I've changed the docstring as well

**Review by adverma98 — DISMISSED** [review]
lgtm!

File: githubmarkdownui/blocks/leaf.py [hunk 3]
```diff
@@ -18,11 +18,14 @@ def code_block(text: str, language: Optional[str] = None) -> str:
...
 
     :param text: The text for the heading
-    :param level: The level of the heading between 1 to 6 inclusive
+    :param level: The level of the heading between 1 and 6 inclusive
 
-    :raises: Exception if the level is not between 1 to 6 inclusive
+    :raises: Exception if the level is not between 1 and 6 inclusive
     """
-    pass
+    if not 1 <= level <= 6:
```
Comment (shajinzheng) [inline githubmarkdownui/blocks/leaf.py]: Can we create constants for both lowest level and highest level?

File: githubmarkdownui/blocks/lists.py (hunk 1 above)
Comment (shajinzheng) [reply githubmarkdownui/blocks/lists.py:21]: Do you think it'd be more clear to document in `HtmlList` that, `__str__` needs to be overridden to output HTML-flavored markdown syntax in inherited classes?

**Review by shajinzheng — DISMISSED** [review]

File: githubmarkdownui/blocks/leaf.py (hunk 3 above)
Comment (TerrenceHung) [reply githubmarkdownui/blocks/leaf.py]: Created `HEADING_MIN_LEVEL = 1` and `HEADING_MAX_LEVEL = 6` in `constants.py`

File: githubmarkdownui/blocks/lists.py (hunk 1 above)
Comment (TerrenceHung) [reply githubmarkdownui/blocks/lists.py:21]: Good idea. I've updated the docstring of `HtmlList` to state that, and also marked `__str__` as an abstract method

**Review by nkamkheaw — APPROVED** [review]
LGTM

**Review by shajinzheng — APPROVED** [review]

**Pull request merged**